src/backend/distributed/utils/citus_ruleutils.c

```diff
@@ -229,51 +229,61 @@ pg_get_tableschemadef_string(Oid tableRelationId)
 	AttrNumber constraintIndex = 0;
 	AttrNumber constraintCount = 0;
 	StringInfoData buffer = { NULL, 0, 0, 0 };
+	OverrideSearchPath *overridePath = GetOverrideSearchPath(CurrentMemoryContext);
 
-	/*
-	 * Instead of retrieving values from system catalogs as other functions in
-	 * ruleutils.c do, we follow an unusual approach here: we open the relation,
-	 * and fetch the relation's tuple descriptor. We do this because the tuple
-	 * descriptor already contains information harnessed from pg_attrdef,
-	 * pg_attribute, pg_constraint, and pg_class; and therefore using the
-	 * descriptor saves us from a lot of additional work.
-	 */
-	relation = relation_open(tableRelationId, AccessShareLock);
-	relationName = generate_relation_name(tableRelationId, NIL);
+	overridePath->schemas = NIL;
+	overridePath->addCatalog = true;
+	PushOverrideSearchPath(overridePath);
 
-	relationKind = relation->rd_rel->relkind;
-	if (relationKind != RELKIND_RELATION && relationKind != RELKIND_FOREIGN_TABLE)
+	PG_TRY();
 	{
-		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
-						errmsg("%s is not a regular or foreign table", relationName)));
-	}
+		/*
+		 * Instead of retrieving values from system catalogs as other functions in
+		 * ruleutils.c do, we follow an unusual approach here: we open the relation,
+		 * and fetch the relation's tuple descriptor. We do this because the tuple
+		 * descriptor already contains information harnessed from pg_attrdef,
+		 * pg_attribute, pg_constraint, and pg_class; and therefore using the
+		 * descriptor saves us from a lot of additional work.
+		 */
+		relation = relation_open(tableRelationId, AccessShareLock);
+		relationName = generate_relation_name(tableRelationId, NIL);
 
-	initStringInfo(&buffer);
-	if (relationKind == RELKIND_RELATION)
-	{
-		appendStringInfo(&buffer, "CREATE TABLE %s (", relationName);
-	}
-	else
-	{
-		appendStringInfo(&buffer, "CREATE FOREIGN TABLE %s (", relationName);
-	}
+		relationKind = relation->rd_rel->relkind;
+		if (relationKind != RELKIND_RELATION && relationKind != RELKIND_FOREIGN_TABLE)
+		{
+			ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
+							errmsg("%s is not a regular or foreign table", relationName)));
```

Comment:
Could you run citus_indent? We added PG_TRY() blocks, so some lines are longer than 90 characters now.